### ISSUE TITLE
[crypto-autotest] qat_setup.sh: line 148: /sys/bus/pci/drivers/dh895xccvf/unbind: Permission denied

### ISSUE THREAD

**Issue by apaszkie** [spdk/spdk#3045]
# CI Intermittent Failure

crypto-autotest fails on CYP13 in qat_setup.sh

## Link to the failed CI build

https://ci.spdk.io/results/autotest-per-patch/builds/105596/archive/crypto-autotest/build.log

## Execution failed at

```
00:05:36.974   16:07:45	-- spdk/autotest.sh@156 -- # /var/jenkins/workspace/crypto-autotest/spdk/scripts/qat_setup.sh
00:05:37.546  Restarting all devices.
00:05:37.546  lstat() error: No such file or directory
00:05:37.546  QAT Error: No GENERAL section found
00:05:37.546  Failed to configure qat_dev0
00:05:37.546  enable sriov
00:05:37.546  Checking status of all devices.
00:05:37.546  There is 1 QAT acceleration device(s) in the system:
00:05:37.546   qat_dev0 - type: dh895xcc,  inst_id: 0,  node_id: 0,  bsf: 0000:4d:00.0,  #accel: 6 #engines: 12 state: down
00:05:38.489  0000:4d:00.0 set to 32 VFs
00:05:38.489  unbinding driver dh895xccvf from qat VF at BDF 0000:4d:03.0
00:05:38.489  /var/jenkins/workspace/crypto-autotest/spdk/scripts/qat_setup.sh: line 148: /sys/bus/pci/drivers/dh895xccvf/unbind: Permission denied
00:05:38.489    16:07:47	-- spdk/autotest.sh@156 -- # trap - ERR
00:05:38.489    16:07:47	-- spdk/autotest.sh@156 -- # print_backtrace
00:05:38.489    16:07:47	-- common/autotest_common.sh@1107 -- # [[ ehxBET =~ e ]]
00:05:38.489    16:07:47	-- common/autotest_common.sh@1109 -- # args=('/var/jenkins/workspace/crypto-autotest/autorun-spdk.conf')
00:05:38.489    16:07:47	-- common/autotest_common.sh@1109 -- # local args
00:05:38.489    16:07:47	-- common/autotest_common.sh@1111 -- # xtrace_disable
00:05:38.489    16:07:47	-- common/autotest_common.sh@20 -- # '[' no '!=' yes ']'
00:05:38.489    16:07:47	-- common/autotest_common.sh@21 -- # PREV_BASH_OPTS=ehxBET
00:05:38.489    16:07:47	-- common/autotest_common.sh@22 -- # [[ ehxBET == *\x* ]]
00:05:38.489    16:07:47	-- common/autotest_common.sh@23 -- # XTRACE_DISABLED=yes
00:05:38.489    16:07:47	-- common/autotest_common.sh@25 -- # set +x
00:05:38.490  ========== Backtrace start: ==========
00:05:38.490  
00:05:38.490  in /var/jenkins/workspace/crypto-autotest/spdk/autotest.sh:156 -> main(["/var/jenkins/workspace/crypto-autotest/autorun-spdk.conf"])
00:05:38.490       ...
00:05:38.490     151 	if [ $SPDK_RUN_FUNCTIONAL_TEST -eq 1 ]; then
00:05:38.490     152 		if [[ $SPDK_TEST_CRYPTO -eq 1 || $SPDK_TEST_VBDEV_COMPRESS -eq 1 ]]; then
00:05:38.490     153 			if [[ $SPDK_TEST_USE_IGB_UIO -eq 1 ]]; then
00:05:38.490     154 				$rootdir/scripts/qat_setup.sh igb_uio
00:05:38.490     155 			else
00:05:38.490  => 156 				$rootdir/scripts/qat_setup.sh
00:05:38.490     157 			fi
00:05:38.490     158 		fi
00:05:38.490     159 		timing_enter lib
00:05:38.490     160 	
00:05:38.490     161 		run_test "env" $rootdir/test/env/env.sh
00:05:38.490       ...
00:05:38.490  
00:05:38.490  ========== Backtrace end ==========
```

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @apaszkie. log: https://ci.spdk.io/public_build/autotest-per-patch_105588.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @apaszkie. log: https://ci.spdk.io/public_build/autotest-per-patch_105596.html

**Comment by mikeBashStuff** [comment]
All reports point at CYP13. It seems to be happening at random (sometimes the setup fails, sometimes it succeeds). Not sure if it relates to faulty QAT or if the setup itself is faulty. I will disable this node and troubleshoot further.

**Comment by mikeBashStuff** [comment]
https://review.spdk.io/gerrit/c/spdk/spdk/+/18527, tested under CYP13. There's a small race when the VFs start to pop up for the first time on the pci bus - when we try to bind them to the uio driver, we happen to do this while the qat_*vf driver is still initializing the device so kernel rejects our write to /unbind. This patch should prevent that from happening.

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @apaszkie. log: https://ci.spdk.io/public_build/autotest-per-patch_105608.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @apaszkie. log: https://ci.spdk.io/public_build/autotest-per-patch_105610.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @tomzawadzki. log: https://ci.spdk.io/public_build/autotest-per-patch_105549.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @KKaras169169. log: https://ci.spdk.io/public_build/autotest-per-patch_105818.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @KKaras169169. log: https://ci.spdk.io/public_build/autotest-per-patch_105816.html

**Comment by KKaras169169** [comment]
> Another instance of this failure. Reported by @KKaras169169. log: https://ci.spdk.io/public_build/autotest-per-patch_105818.html

> Another instance of this failure. Reported by @KKaras169169. log: https://ci.spdk.io/public_build/autotest-per-patch_105816.html

These two were reported on backports to v23.01.x branch. These logs slighty differ from posted backtrace, but I believe this is a similar problem.

**Comment by tomzawadzki** [comment]
[Bug scrub] The latest reports were from the v23.01.x LTS branch that will shortly contain backport of the relevant patch. If the issue continues to be no longer present on master SPDK, it can be closed.

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @KKaras169169. log: https://ci.spdk.io/public_build/autotest-per-patch_105908.html

**Comment by tomzawadzki** [comment]
[Bug scrub] The patch was merged to latest SPDK and backported to v23.01.x and v23.05.x branches. Thank you @mikeBashStuff !
